src/algorithms/a_star_algorithm.py

```diff
@@ -0,0 +1,128 @@
+from dataclasses import dataclass
+from typing import List, Tuple
+
+from lynx.common.vector import Vector
+
+
+class Node:
+    def __init__(self, parent=None, position=None):
+        self.parent = parent
+        self.position = position
+
+        self.g = 0
+        self.h = 0
+        self.f = 0
+
+    def __eq__(self, other):
+        return self.position == other.position
+
+
+@dataclass
+class AStarAlgorithm:
+    """
+    Get the shortest path from start to end
+    """
+    start: Vector = None
+    end: Vector = None
+
+    def get_path(self, scene: 'Scene') -> List[Vector]:

```

Comment:
There's a HUGE opportunity to use helper functions here.
Consider it a rule of thumb, that if you need to write a comment for a fragment of code, this fragment of code can become a function.

For instance, this:
```
# found the goal
            if current_node == end_node:
                path = []
                current = current_node
                while current is not None:
                    path.append(current.position)
                    current = current.parent
                return self.transform_to_unit_vectors(path[::-1])
```
can become this:
```
if path := is_goal_found(current_node, end_node):
    return path
```

Reply:
Okay, I will keep this in mind in the future, for now I would leave it as is. On the other hand:
```
if path := is_goal_found(current_node, end_node):
    return path
```
`is_goal_found` suggests that a boolean value will be returned